pkg/controller/managedresource/add.go

```diff
@@ -137,3 +155,26 @@ func (o *ControllerOptions) Completed() *ControllerConfig {
 func (c *ControllerConfig) ApplyClassFilter(filter *filter.ClassFilter) {
 	*filter = *c.ClassFilter
 }
+
+// determineClusterIdentity is used to extract the cluster identity from the cluster-identity
+// config map. This is intended as fallback if no explicit cluster identity is given.
+// in  seed-shoot scenario, the cluster id for the managed resources must be explicitly given
+// to support the migration of a shoot from one seed to another. Here the identity `seed` should
+// be set.
+func determineClusterIdentity(c client.Client, force bool) (string, error) {
+	cm := corev1.ConfigMap{}
+	err := c.Get(context.Background(), client.ObjectKey{Name: "cluster-identity", Namespace: "kube-system"}, &cm)
```

Comment:
```suggestion
	err := c.Get(context.Background(), client.ObjectKey{Name: v1beta1constants.ClusterIdentity, Namespace: metav1.NamespaceSystem}, &cm)
```

From https://github.com/gardener/gardener/blob/3c3c53d0f53f537c8411468ace0b77d560aca751/pkg/apis/core/v1beta1/constants/types_constants.go#L324